consensus/src/execution_ctx.rs

```diff
@@ -126,21 +135,46 @@ impl<'a, DB: Database, T: Operations + 'static> ExecutionCtx<'a, DB, T> {
         &mut self,
         phase: Arc<Mutex<C>>,
     ) -> Result<Message, ConsensusError> {
-        debug!(event = "run event_loop");
+        let open_consensus_mode = self.last_step_running();
+
+        // When consensus is in open_consensus_mode then it keeps Ratification
+        // step running infinitely until either a valid block or
+        // emergency block is accepted
+        let timeout = if open_consensus_mode {
+            let dur = Duration::new(u32::MAX as u64, 0);
+            info!(event = "run event_loop in open_mode", ?dur);
+            dur
+        } else {
+            let dur = self.iter_ctx.get_timeout(self.step_name());
+            debug!(event = "run event_loop", ?dur);
+            dur
+        };
 
-        let timeout = self.iter_ctx.get_timeout(self.step_name());
         let deadline = Instant::now().checked_add(timeout).unwrap();
-
         let inbound = self.inbound.clone();
 
         // Handle both timeout event and messages from inbound queue.
         loop {
             match time::timeout_at(deadline, inbound.recv()).await {
                 // Inbound message event
                 Ok(Ok(msg)) => {
+                    let success_quorum = match &msg.payload {
+                        Payload::Quorum(q) => matches!(
+                            q.att.result,
+                            RatificationResult::Success(Vote::Valid(_))
+                        ),
+                        _ => false,
+                    };
+
                     if let Some(step_result) =
                         self.process_inbound_msg(phase.clone(), msg).await
                     {
+                        if open_consensus_mode && !success_quorum {
+                            // In open consensus mode, consensus step is
+                            // terminated only in case of a success quorum
+                            continue;
+                        }
+
```

Comment:
I'm not sure about this for the following reasons:
- Having a success Quorum doesn't mean we should stop the consensus, it should be terminated after the block is accepted
- What if the quorum refers to a previous iteration collected during the emerency mode? Returning here would increase the current iteration

Reply:
According to the current specifications, the emergency mode runs until a Success Quorum is reached (or an Emergency Block). That is, we conclude the round as soon as a valid block for the round has been produced. So that part I think it's correct, unless I'm missing something in your comment.

On the other hand, I now have my own doubts on whether Quorum messages are handled properly by `process_inbound_msg`... AFAICS, the function only takes step messages into account, returning Some if a Ratification messages reach a quorum.

Reply:
> * Having a success Quorum doesn't mean we should stop the consensus, it should be terminated after the block is accepted

As discussed, I've changed the open_consensus_mode to ignore quorum.

> * What if the quorum refers to a previous iteration collected during the emerency mode? Returning here would increase the current iteration

As analyzed, the process_from_past never return `Some`.
As discussed I changed to code to explicitly return `None`